source/DSCResources/DSC_CMBoundaryGroups/DSC_CMBoundaryGroups.psm1

```diff
@@ -0,0 +1,364 @@
+
+        [Parameter()]
+        [String[]]
+        $BoundariesToExclude,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -BoundaryGroup $BoundaryGroup
+    $result = $true
+
+    if ($Ensure -eq 'Present')
+    {
+        if ($state.Ensure -eq 'Absent')
+        {
+            Write-Verbose -Message ($script:localizedData.BoundaryGroupMissing -f $BoundaryGroup)
+            $result = $false
+        }
+        else
+        {
+            if (($Boundaries) -or ($BoundariesToInclude))
+            {
+                if ($Boundaries)
+                {
+                    $includes = $Boundaries
+                }
+                else
+                {
+                    $includes = $BoundariesToInclude
+                }
+
+                foreach ($member in $includes)
+                {
+                    if ($state.Boundaries -notcontains $member)
+                    {
+                        Write-Verbose -Message ($script:localizedData.MissingBoundary -f $BoundaryGroup, $member)
+                        $result = $false
+                    }
+                }
+            }
+
+            if (($Boundaries) -or ($BoundariesToExclude))
+            {
+                if (-not [string]::IsNullOrEmpty($state.Boundaries))
+                {
+                    if ($boundaries)
+                    {
+                        $excludes = $Boundaries
+                    }
+                    else
+                    {
+                        $excludes = $BoundariesToExclude
+                    }
+
+                    $compares = Compare-Object -ReferenceObject $state.Boundaries -DifferenceObject $excludes -IncludeEqual
+                    foreach ($compare in $compares)
+                    {
+                        if ($Boundaries)
+                        {
+                            if ($compare.SideIndicator -eq '<=')
+                            {
+                               Write-Verbose -Message ($script:localizedData.ExtraBoundary `
+                                    -f $BoundaryGroup, $compare.InputObject)
+                                $result = $false
+                            }
+                        }
+                        else
+                        {
+                            if ($compare.SideIndicator -eq '==')
+                            {
+                                Write-Verbose -Message ($script:localizedData.ExtraBoundary `
+                                    -f $BoundaryGroup, $compare.InputObject)
+                                $result = $false
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    else
+    {
+        if ($state.Ensure -eq 'Present')
```

Comment:
You can make the else an elseif:
elseif (($Ensure -eq 'Present') -and ($state.Ensure -eq 'Present'))